torch/onnx/_internal/exporter.py

```diff
@@ -75,14 +82,179 @@ class ONNXFakeContext:
+        from onnxscript.function_libs.torch_lib import (  # type: ignore[import]  # noqa: F401
+            ops,  # TODO(titaiwang): get rid of this import
+            registration,
+        )
+
+        # opset_version is unused for now, since torchlib only supports opset18.
+        self._opset_version = _DEFAULT_OPSET_VERSION
+        warnings.warn(
+            f"Torchlib only supports opset version {self._opset_version} for now. If you need to use a \
+            different opset version, please register them with register_custom_op."
+        )
+
+        # Initialize registry from torchlib
+        self._initiate_registry_from_torchlib(registration.default_registry)
+
+    @property
+    def opset_version(self) -> int:
+        """The ONNX opset version the exporter should target. Defaults to the latest
+        supported ONNX opset version. The default version will increment over time as
+        ONNX continues to evolve."""
+
+        return self._opset_version
+
+    # TODO(titaiwang): subject to change if multiple opset_version is supported in torchlib
+    def _initiate_registry_from_torchlib(
+        self, torchlib_registry: torchlib_registry.Registry
+    ):
+        """Populates the registry with ATen functions from torchlib.
+
+        Args:
+            torchlib_registry: The torchlib registry to use for populating the registry.
+        """
+        for aten_name, aten_overloads_func in torchlib_registry.items():
+            internal_name_instance = registration.OpName.from_qualified_name(aten_name)
+            for overload_func in aten_overloads_func.overloads:
+                symbolic_function = registration.SymbolicFunction(
+                    onnx_function=overload_func,
+                    op_full_name=internal_name_instance.qualified_name(),
+                    is_custom=False,
+                    is_complex=False,
+                )
+                self._register(internal_name_instance, symbolic_function)
+
+            for complex_func in aten_overloads_func.complex:
+                symbolic_function = registration.SymbolicFunction(
+                    onnx_function=complex_func,
+                    op_full_name=internal_name_instance.qualified_name(),
+                    is_custom=False,
+                    is_complex=True,
+                )
+                self._register(internal_name_instance, symbolic_function)
+
+    @_beartype.beartype
+    def _register(
+        self,
+        internal_qualified_name: registration.OpName,
+        symbolic_function: registration.SymbolicFunction,
+    ) -> None:
+        """Registers a SymbolicFunction to an operator.
+
+        Args:
+            internal_qualified_name: The qualified name of the operator to register: OpName.
+            symbolic_function: The SymbolicFunction to register.
+        """
+        self._registry[internal_qualified_name].append(symbolic_function)
+
+    @_beartype.beartype
+    def register_custom_op(
```

Comment:
```suggestion
    def register_op(
```
As discussed in during meeting, ideally this API could be used to register custom operators, but also to override existing ones, which would make the "custom" word confusing